pkg/apis/peakq/v0alpha1/types.go

```diff
@@ -37,43 +37,66 @@ type Target struct {
 	// TODO 2[uint] seems to panic, maybe implement DeepCopy on data.FrameTypeVersion?
 	// DataTypeVersion data.FrameTypeVersion `json:"dataTypeVersion,omitempty"`
 
+	// Variables that will be replaced in the query
+	Variables map[string][]VariableReplacement `json:"variables"`
+
+	// The raw query: TODO, should be query.GenericQuery
 	Properties common.Unstructured `json:"properties"`
 }
 
-// QueryVariable is the definition of a variable that will be interpolated
+// TemplateVariable is the definition of a variable that will be interpolated
 // in targets.
-type QueryVariable struct {
+type TemplateVariable struct {
 	// Key is the name of the variable.
 	Key string `json:"key"`
 
 	// DefaultValue is the value to be used when there is no selected value
 	// during render.
-	DefaultValue string `json:"defaultValue"`
-
-	// Positions is a list of where to perform the interpolation
-	// within targets during render.
-	// The first string is the Idx of the target as a string, since openAPI
-	// does not support ints as map keys
 	// +listType=atomic
-	//Positions map[string]map[Path][]Position `json:"positions"`
-	Positions []Position `json:"positions"`
+	DefaultValues []string `json:"defaultValue"`
 
 	// ValueListDefinition is the object definition used by the FE
 	// to get a list of possible values to select for render.
 	ValueListDefinition common.Unstructured `json:"valueListDefinition"`
 }
 
-// Position is where to do replacement in the targets
-// during render.
-type Position struct {
-	// IndexIdx is the index of the target in The QueryTemplateSpec Targets property.
-	TargetIdx int `json:"targetIdx"`
-
-	// TargetKey is the location of the property within the the target properties.
+// QueryVariable is the definition of a variable that will be interpolated
+// in targets.
+type VariableReplacement struct {
+	// Path is the location of the property within the the target properties.
 	// The format for this is not figured out yet (Maybe JSONPath?).
 	// Idea: ["string", int, "string"] where int indicates array offset
 	Path string `json:"path"`
 
+	// Positions is a list of where to perform the interpolation
+	// within targets during render.
+	// The first string is the Idx of the target as a string, since openAPI
+	// does not support ints as map keys
+	Position *Position `json:"position,omitempty"`
+
+	// How values should be interpolated
+	// See: https://grafana.com/docs/grafana/latest/dashboards/variables/variable-syntax/#advanced-variable-format-options
+	// NOTE: the format parameter is not yet supported!
+	Format string `json:"format,omitempty"`
```

Comment:
^^ for now, we should throw an error if this is configured -- the options mostly support what todo with multi-valued fields